fastlane/lanes/build.rb

```diff
@@ -127,18 +127,29 @@
   # Builds the WordPress app and uploads it to TestFlight, for beta-testing or final release
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
-  # @option [Boolean] skip_prechecks (default: false) If true, don't run the ios_build_prechecks and ios_build_preflight
+  # @option [Boolean] skip_prechecks (default: false) If true, don't run the prechecks and ios_build_preflight
   # @option [Boolean] create_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
   # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
   #
   # @called_by CI
   #
   desc 'Builds and uploads for distribution to App Store Connect'
   lane :build_and_upload_app_store_connect do |options|
-    ios_build_prechecks(skip_confirm: options[:skip_confirm], external: true) unless options[:skip_prechecks]
-    ios_build_preflight unless options[:skip_prechecks]
-
     sentry_check_cli_installed
+
+    unless options[:skip_prechecks]
+      message = "Building version #{current_release_version} (#{current_build_code}) and uploading to TestFlight\n"
+      if options[:skip_confirm]
+        UI.message(message)
+      else
+        UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
```

Comment:
Have you considered adding the \n in the UI.confirm only? UI.message should already handle separating messages in new lines.

```suggestion
      message = "Building version #{current_release_version} (#{current_build_code}) and uploading to TestFlight."
      if options[:skip_confirm]
        UI.message(message)
      else
        UI.user_error!('Aborted by user request') unless UI.confirm("#{message}\nDo you want to continue?")
```

Reply:
See https://github.com/wordpress-mobile/WordPress-iOS/pull/21666#discussion_r1350991187